tests/Operation/WatchFunctionalTest.php

```diff
@@ -1190,6 +1197,266 @@ public function testSessionFreed()
+        $this->assertEquals($aggregateCommands[0], $aggregateCommands[1]);
+    }
+
+    /**
+     * Prose test: "ChangeStream will not attempt to resume on any error
+     * encountered while executing an aggregate command."
+     */
+    public function testErrorDuringAggregateCommandDoesNotCauseResume()
+    {
+        if (version_compare($this->getServerVersion(), '4.0.0', '<')) {
+            $this->markTestSkipped('failCommand is not supported');
+        }
+
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], $this->defaultOptions);
+
+        $commandCount = [];
+
+        $this->configureFailPoint([
+            'configureFailPoint' => 'failCommand',
+            'mode' => ['times' => 1],
+            'data' => ['failCommands' => ['aggregate'], 'errorCode' => self::NOT_MASTER],
+        ]);
+
+        $this->expectException(CommandException::class);
+
+        (new CommandObserver)->observe(
+            function() use ($operation) {
+                $operation->execute($this->getPrimaryServer());
+            },
+            function(array $event) use (&$commandCount) {
+                $commandCount++;
+            }
+        );
+
+        $this->assertSame(1, $commandCount);
+    }
+
+    /**
+     * Prose test: "ChangeStream will perform server selection before attempting
+     * to resume, using initial readPreference"
+     */
+    public function testOriginalReadPreferenceIsPreservedOnResume()
+    {
+        $readPreference = new ReadPreference(ReadPreference::RP_SECONDARY);
+        $options = ['readPreference' => $readPreference] + $this->defaultOptions;
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], $options);
+
+        $secondary = $this->manager->selectServer($readPreference);
+
+        $changeStream = $operation->execute($secondary);
+        $previousCursorId = $changeStream->getCursorId();
+        $this->killChangeStreamCursor($changeStream);
+
+        $changeStream->next();
+        $this->assertNotSame($previousCursorId, $changeStream->getCursorId());
+
+        $getCursor = Closure::bind(
+            function () {
+                return $this->iterator->getInnerIterator();
+            },
+            $changeStream,
+            ChangeStream::class
+        );
+        /** @var Cursor $cursor */
+        $cursor = $getCursor();
+        self::assertTrue($cursor->getServer()->isSecondary());
+    }
+
+    /**
+     * Prose test
+     * For a ChangeStream under these conditions:
+     * - Running against a server <4.0.7.
+     * - The batch is empty or has been iterated to the last document.
+     * Expected result:
+     * - getResumeToken must return the _id of the last document returned if one exists.
+     * - getResumeToken must return resumeAfter from the initial aggregate if the option was specified.
+     * - If resumeAfter was not specified, the getResumeToken result must be empty.
```

Comment:
I see the first two assertions in the test, but is this third case being tested?

Reply:
That is checked in the first assertion to `$changeStream->getResumeToken()` directly after executing the `watch` operation. It was easier testing that directly after running the operation since we have a change stream with an empty batch and without a `resumeAfter` option.

Reply:
Ah, I missed that line. LGTM!